coops-ui/tests/unit/Certify.spec.ts

```diff
@@ -0,0 +1,172 @@
+//
+// Copyright © 2019 Province of British Columbia
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+//
+
+import Vue from 'vue'
+import Vuetify from 'vuetify'
+
+import { mount, Wrapper } from '@vue/test-utils'
+
+import Certify from '@/components/AnnualReport/Certify.vue'
+
+Vue.use(Vuetify)
+
+// Boilerplate to prevent the complaint "[Vuetify] Unable to locate target [data-app]"
+const app: HTMLDivElement = document.createElement('div')
+app.setAttribute('data-app', 'true')
+document.body.append(app)
+
+// Input field selectors to test changes to the DOM elements.
+const certifiedBySelector: string = 'input[type=text]'
+const isCertifiedSelector: string = 'input[type=checkbox]'
+const statementSelector: string = '.certify-stmt'
+
+const someCertifier = 'Some Certifier'
+const defaultDate = '2019-01-01'
+
+/**
+ * Returns the last event for a given name, to be used for testing event propagation in response to component changes.
+ *
+ * @param wrapper the wrapper for the component that is being tested.
+ * @param name the name of the event that is to be returned.
+ *
+ * @return the value of the last named event for the wrapper.
+ */
+function getLastEvent (wrapper: Wrapper<Certify>, name: string): any {
+  const eventsList: Array<any> = wrapper.emitted(name)
+  const events: Array<any> = eventsList[eventsList.length - 1]
```

Comment:
What happens if an event was never emitted? :P

Reply:
Good question - "it explodes" is the answer.

The tests always test for existence before calling getLastEvent,, but why not just let it fail? The resulting stack trace is enough information to find the caller and determine which event is missing.